### PR TITLE
New create site API for Marketplace:

### PR DESCRIPTION
**Pull request by jdrossl**
- Marketplace service handles validations and triggers the create

Depends on https://github.com/craftercms/commons/pull/192

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -3025,6 +3025,35 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /marketplace/create_site:
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Why not `/site/create_site_from_marketplace`?

`/site/` becomes the new root for API2 site operations.

My issue with `/marketplace` is that it's misleading thinking this is the marketplace of Studio when it's just create site API specialized.

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java [hunk 2]
```diff
@@ -149,4 +181,52 @@ protected void validate() throws MarketplaceException {
         }
     }
 
+    public Map<String, Object> getDescriptor(String id, Version version) throws MarketplaceException,
```
Comment (dejan-brkic) [inline src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java]: Public methods of v2 services should be protected with permissions

File: src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java [hunk 3]
```diff
@@ -149,4 +181,52 @@ protected void validate() throws MarketplaceException {
...
+        } catch (ResourceAccessException e) {
+            throw new MarketplaceUnreachableException(url, e);
+        } catch (IllegalArgumentException e) {
+            throw new BlueprintNotFoundException(String.format("Blueprint not found in the Marketplace: %s %s.%s.%s",
+             id, version.getMajor(), version.getMinor(), version.getPatch()));
+        }
+    }
+
+    @Override
+    public void createSite(CreateSiteRequest request) throws RemoteRepositoryNotFoundException,
```
Comment (dejan-brkic) [inline src/main/java/org/craftercms/studio/impl/v2/service/marketplace/MarketplaceServiceImpl.java:68]: Public methods of v2 services should be protected with permissions

**Review by dejan-brkic — CHANGES_REQUESTED** [review]

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml]: I added it there to call directly the marketplace service and avoid having to change the other service, my idea was to wrap the site service so they continue to be independent (mp can change without changing site service and the other way around)

Comment (jdrossl) [reply src/main/api/2/studio-api-2.yaml]: If you are only worried about the URL then we can move it to the `SitesController` and still call the MP service. Let me know if that works or if you want to talk about it

**Review by dejan-brkic — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**